src/picker/fzf.rs

```diff
@@ -0,0 +1,76 @@
+use crate::error;
+use crate::models;
+use crate::picker;
+use error::PickerError;
+use itertools::Itertools;
+use models::ResultWithDefaultError;
+use picker::{ItemPicker, PickableItem};
+use std::collections::HashMap;
+use std::io;
+use std::io::Write;
+use std::process::{Command, Stdio};
+
+pub struct FzfPicker;
+
+fn remove_trailing_newline(value: String) -> String {
+    let mut chars = value.chars();
+    chars.next_back();
+    chars.as_str().to_string()
+}
+
+fn format_as_fzf_input(items: &[PickableItem]) -> String {
+    items
+        .iter()
+        .map(|item| item.formatted.clone())
+        .unique()
+        .fold("".to_string(), |acc, item| acc + item.as_str() + "\n")
+}
+
+fn create_element_hash_map(items: &[PickableItem]) -> HashMap<String, i64> {
+    items
+        .iter()
+        .map(|item| (item.formatted.clone(), item.id))
+        .collect::<HashMap<String, i64>>()
+}
+
+impl ItemPicker for FzfPicker {
+    fn pick(&self, items: Vec<PickableItem>) -> ResultWithDefaultError<i64> {
+        let mut command = Command::new("fzf");
+        command
+            .arg("-n2..")
+            .arg("--ansi")
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped());
+
+        match command.spawn() {
+            Ok(mut child) => {
+                let fzf_input = format_as_fzf_input(&items);
+                let possible_elements = create_element_hash_map(&items);
+
+                writeln!(child.stdin.as_mut().unwrap(), "{}", fzf_input)?;
+
+                match child.wait_with_output() {
+                    Err(_) => Err(Box::new(PickerError::Generic)),
+                    Ok(output) => match output.status.code() {
+                        Some(0) => {
+                            let user_selected_string = String::from_utf8(output.stdout)?;
+                            println!("{}", user_selected_string);
+                            let selected_item_index = remove_trailing_newline(user_selected_string);
+                            println!("{}", selected_item_index);
+                            let selected_item =
+                                possible_elements.get(&selected_item_index).unwrap();
+                            Ok(*selected_item)
+                        }
+
+                        Some(128..=254) | None => Err(Box::new(PickerError::Cancelled)),
+                        _ => Err(Box::new(PickerError::Generic)),
```

Comment:
I reached here but filtering for a non-existenent entry, the issue was the error logged just showed the name for the enum instead of the actual message.
![image](https://user-images.githubusercontent.com/1716853/163685742-3d49e168-aeb1-4501-a426-07550820beae.png)

Reply:
Great catch! We have pretty printed errors but we are not actually showing any of them. I've just fixed this!